src/ffi.rs

```diff
@@ -147,6 +148,43 @@ pub unsafe extern "C" fn public_spend_key(
     0
 }
 
+/// Execute a generic contract call
+#[no_mangle]
+pub unsafe extern "C" fn execute(
+    contract_id: *const [u8; 32],
+    call_data_ptr: *mut u8,
+    call_data_len: *const u32,
+    sender_index: *const u64,
+    refund: *const [u8; PublicSpendKey::SIZE],
+    gas_limit: *const u64,
+    gas_price: *const u64,
+) -> u8 {
+    let contract_id = ContractId::from(*contract_id);
+
+    // SAFETY: the buffer is expected to have been allocated with the
+    // correct size. If this is not the case problems with the allocator
+    // *may* happen.
```

Comment:
Should this be in the method comment too?

Reply:
I was just be consistent with the rest of the code base.
Keep in mind that, where `SAFETY` is absolutely needed for documenting `unsafe` function, is not for FFI – since they're all unsafe.
My guessing is that @ureeves just wanted to highlight the exact method call where this is happening, more for maintenance that actually documentation.